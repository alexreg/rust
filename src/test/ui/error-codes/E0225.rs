#![feature(trait_alias)]

trait Foo = std::io::Read + std::io::Write;
trait Bar = Foo;

fn main() {
    let _: Box<std::io::Read + std::io::Write>;
    //~^ ERROR only auto traits can be used as additional traits in a trait object [E0225]
    let _: Box<Bar>;
    //~^ ERROR only auto traits can be used as additional traits in a trait object [E0225]
}
