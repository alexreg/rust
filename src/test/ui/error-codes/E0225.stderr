error[E0225]: only auto traits can be used as additional traits in a trait object
  --> $DIR/E0225.rs:7:32
   |
LL |     let _: Box<std::io::Read + std::io::Write>;
   |                                ^^^^^^^^^^^^^^ non-auto additional trait

error[E0225]: only auto traits can be used as additional traits in a trait object
  --> $DIR/E0225.rs:9:16
   |
LL | trait Foo = std::io::Read + std::io::Write;
   |                             -------------- non-auto additional trait
LL | trait Bar = Foo;
   |             --- referenced here
...
LL |     let _: Box<Bar>;
   |                ^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0225`.
